metals/src/main/scala/scala/meta/internal/rename/RenameProvider.scala

```diff
@@ -67,224 +67,235 @@ final class RenameProvider(
       token: CancelToken,
   ): Future[Option[LSPRange]] = {
     val source = params.getTextDocument.getUri.toAbsolutePath
-    compilations.compilationFinished(source).flatMap { _ =>
-      definitionProvider.definition(source, params, token).map { definition =>
-        val symbolOccurrence: Option[(SymbolOccurrence, TextDocument)] =
-          definitionProvider
-            .symbolOccurrence(source, params.getPosition)
-            .orElse(
-              findRenamedImportOccurrenceAtPosition(
-                source,
-                params.getPosition(),
-              )
-            )
-        for {
-          (occurence, _) <- symbolOccurrence
-          soughtSymbols = Set(occurence.symbol) ++ companion(occurence.symbol)
-          definitionLocation <- definition.locations.asScala.headOption
-          definitionPath = definitionLocation.getUri().toAbsolutePath
-          if canRenameSymbol(occurence.symbol, None) &&
-            (isWorkspaceSymbol(occurence.symbol, definitionPath) ||
-              findRenamedImportForSymbol(
-                source,
-                soughtSymbols,
-                occurence.symbol.desc.name.value,
-              ).isDefined)
-          range <- occurence.range
-        } yield range.toLsp
+    val localPrepareRename =
```

Comment:
Actually, maybe it does make sense to only use prepare rename in the compiler? We don't really need semanticdb here yet.

Reply:
But when symbol is not local, we will use SemanticDB for rename. We can't check here if all of the files compile and semanticdb won't work there. The effect would be that the prompt for new name will show, but rename will have no effect (is it that bad actually?)